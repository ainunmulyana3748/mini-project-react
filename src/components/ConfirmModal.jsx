import { AlertTriangle } from "lucide-react";

const ConfirmModal = ({
  type = "delete",
  title,
  message,
  onConfirm,
  onCancel,
}) => {
  const isDelete = type === "delete";

  const iconColor = isDelete ? "text-red-500" : "text-blue-500";
  const buttonColor = isDelete
    ? "bg-red-500 hover:bg-red-600"
    : "bg-blue-500 hover:bg-blue-600";
  const confirmText = isDelete ? "Yes, Delete" : "Yes, Save";

  return (
    <div className="fixed inset-0 bg-black/30 backdrop-blur-sm flex items-center justify-center z-50">
      <div className="bg-white rounded-2xl p-6 shadow-xl text-center w-[320px] animate-fade-in-up">
        <AlertTriangle className={`${iconColor} mx-auto mb-4`} size={48} />
        <h3 className="text-lg font-semibold text-gray-800">{title}</h3>
        <p className="text-sm text-gray-600 mt-1">{message}</p>
        <div className="mt-6 flex justify-center gap-4">
          <button
            className="px-4 py-2 text-sm rounded-lg bg-gray-200 hover:bg-gray-300 text-gray-700 cursor-pointer"
            onClick={onCancel}
          >
            Cancel
          </button>
          <button
            className={`px-4 py-2 text-sm rounded-lg text-white cursor-pointer ${buttonColor}`}
            onClick={onConfirm}
          >
            {confirmText}
          </button>
        </div>
      </div>
    </div>
  );
};

export default ConfirmModal;
