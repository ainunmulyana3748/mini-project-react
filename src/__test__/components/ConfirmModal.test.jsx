import { render, screen, fireEvent } from "@testing-library/react";
import ConfirmModal from "../../components/ConfirmModal";

describe("ConfirmModal Component", () => {
  const defaultProps = {
    title: "Konfirmasi",
    message: "Apakah Anda yakin?",
    onConfirm: jest.fn(),
    onCancel: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("render dengan type delete (default)", () => {
    render(<ConfirmModal {...defaultProps} />);
    expect(screen.getByText("Konfirmasi")).toBeInTheDocument();
    expect(screen.getByText("Apakah Anda yakin?")).toBeInTheDocument();
    expect(screen.getByText("Yes, Delete")).toBeInTheDocument();
    expect(screen.getByText("Cancel")).toBeInTheDocument();
  });

  test("render dengan type edit", () => {
    render(<ConfirmModal {...defaultProps} type="edit" />);
    expect(screen.getByText("Yes, Save")).toBeInTheDocument();
  });

  test("klik tombol Cancel memanggil onCancel", () => {
    render(<ConfirmModal {...defaultProps} />);
    fireEvent.click(screen.getByText("Cancel"));
    expect(defaultProps.onCancel).toHaveBeenCalled();
  });

  test("klik tombol Confirm memanggil onConfirm", () => {
    render(<ConfirmModal {...defaultProps} />);
    fireEvent.click(screen.getByText("Yes, Delete"));
    expect(defaultProps.onConfirm).toHaveBeenCalled();
  });

  test("snapshot confirm modal", () => {
    const { container } = render(<ConfirmModal {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
