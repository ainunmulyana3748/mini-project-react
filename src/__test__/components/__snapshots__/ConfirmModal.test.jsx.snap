// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal Component snapshot confirm modal 1`] = `
<div>
  <div
    class="fixed inset-0 bg-black/30 backdrop-blur-sm flex items-center justify-center z-50"
  >
    <div
      class="bg-white rounded-2xl p-6 shadow-xl text-center w-[320px] animate-fade-in-up"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-triangle-alert text-red-500 mx-auto mb-4"
        fill="none"
        height="48"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m21.73 18-8-14a2 2 0 0 0-3.48 0l-8 14A2 2 0 0 0 4 21h16a2 2 0 0 0 1.73-3"
        />
        <path
          d="M12 9v4"
        />
        <path
          d="M12 17h.01"
        />
      </svg>
      <h3
        class="text-lg font-semibold text-gray-800"
      >
        Konfirmasi
      </h3>
      <p
        class="text-sm text-gray-600 mt-1"
      >
        Apakah Anda yakin?
      </p>
      <div
        class="mt-6 flex justify-center gap-4"
      >
        <button
          class="px-4 py-2 text-sm rounded-lg bg-gray-200 hover:bg-gray-300 text-gray-700 cursor-pointer"
        >
          Cancel
        </button>
        <button
          class="px-4 py-2 text-sm rounded-lg text-white cursor-pointer bg-red-500 hover:bg-red-600"
        >
          Yes, Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
